### PR TITLE
docs: add useSetAtom docs

### LINKED ISSUE #1019: useSetAtom doc search can't find description
When I search `useSetAtom`, no doc page is found.  Searching `useAtomValue` finds a doc page.

### PR DESCRIPTION
**Pull request by Nsttt**
Implements docs for `useSetAtom`.

Closes #1019

### CODE REVIEW

**Review by dai-shi — CHANGES_REQUESTED** [review]
Thanks for working on improving docs!

A few requests for changes:
- `useSetAtom` is not in `jotai/utils` so it shouldn't be in utils.mdx or utils folder.
- It should be in core.mdx, but we need to restructure them to split "Basic APIs" and "Additional APIs" See #885. This might sound out of the scope of this PR, but we need the restructure anyway. Sorry, it hasn't be done yet.
- `useUpdateAtom` docs should be updated saying `useSetAtom` is available. Same for `useAtomValue` in utils. This can be a separate PR.

**Comment by Nsttt** [comment]
Hello !

I've made some of the changes you suggested and they seem to work.
Tell me if there's anything else that can be improved.

File: docs/api/basic.mdx [hunk 1]
```diff
@@ -1,5 +1,5 @@
 ---
-title: Core
+title: Basic
```
Comment (dai-shi) [inline docs/api/basic.mdx:2]: Can we keep core.mdx, and have two sections "Basic APIs" and "Additional APIs" in it?

File: docs/utils/use-update-atom.mdx [hunk 2]
```diff
@@ -4,6 +4,8 @@ title: useUpdateAtom
 
 Ref: https://github.com/pmndrs/jotai/issues/26
 
+For newer versions of Jotai it's highly recomended to use [`useSetAtom`](/docs/api/additional.mdx#use-set-atom) instead.
```
Comment (dai-shi) [inline docs/utils/use-update-atom.mdx:7]: It's not that "highly". In v1, we keep exporting `useUpdateAtom` from `jotai/utils`.

**Review by dai-shi — CHANGES_REQUESTED** [review]

File: docs/api/basic.mdx (hunk 1 above)
Comment (Nsttt) [reply docs/api/basic.mdx]: Sure, done.

File: docs/utils/use-update-atom.mdx (hunk 2 above)
Comment (Nsttt) [reply docs/utils/use-update-atom.mdx]: Alright, makes sense. Done.

File: docs/utils/use-update-atom.mdx [hunk 3]
```diff
@@ -4,6 +4,8 @@ title: useUpdateAtom
 
 Ref: https://github.com/pmndrs/jotai/issues/26
 
+For newer versions of Jotai you can also use [`useSetAtom`](/docs/api/additional.mdx#use-set-atom) instead.
```
Comment (dai-shi) [inline docs/utils/use-update-atom.mdx:7]:
```suggestion
For newer versions of Jotai you can also use [`useSetAtom`](/docs/api/core.mdx#use-set-atom) instead.
```

**Review by dai-shi — APPROVED** [review]
It looks good. Thanks so much for your contribution.

**Pull request merged**